sdk/core/core-lro/src/poller.ts

```diff
@@ -24,10 +24,10 @@ export class PollerCancelledError extends Error {
   }
 }
 
-export interface PollerLike<TState, TResult> {
+export interface PollerLike<TResult> {
   poll(options?: { abortSignal?: AbortSignal }): Promise<void>;
   pollUntilDone(): Promise<TResult>;
-  onProgress(callback: (state: TState) => void): CancelOnProgress;
+  onProgress(callback: (state: PollOperationState<TResult>) => void): CancelOnProgress;
```

Comment:
Hmm I didn't think about this trade-off before. Does this mean that onProgress can't have easy access to intermediate state specific to the operation?

Reply:
Yep, exactly.

Reply:
I have a work item #5801 to support getting the last raw response from LRO operation.  Will that be affected after this change?

Reply:
Probably. I'll close this if this PR is ok: https://github.com/Azure/azure-sdk-for-js/pull/6569